rdtools/system_analysis.py

```diff
@@ -0,0 +1,583 @@
+            'ad_hoc_filter': None  # use this to include an explict filter
+        }
+
+    def calc_clearsky_poa(self, times=None, rescale=True, **kwargs):
+        '''
+        Calculate clearsky plane-of-array irradiance and store in self.clearsky_poa
+
+        Parameters
+        ----------
+        times: Pandas DateTimeIndex for which to calculate clearsky poa
+        rescale: Whether to attempt to rescale clearsky irradiance to measured (bool)
+        model: Model for pvlib.irradiance.get_total_irradiance() (str)
+        kwargs: Extra parameters passed to pvlib.irradiance.get_total_irradiance()
+
+        Returns
+        -------
+        None
+        '''
+        if times is None:
+            times = self.poa.index
+        if self.pvlib_location is None:
+            raise ValueError('pvlib location must be provided')
+        if self.pv_tilt is None or self.pv_azimuth is None:
+            raise ValueError('pv_tilt and pv_azimuth must be provded')
+        if times is not self.poa.index and rescale is True:
+            raise ValueError('rescale=True can only be used when clearsky poa is on same index as poa')
+
+        loc = self.pvlib_location
+        sun = loc.get_solarposition(times)
+        clearsky = loc.get_clearsky(times, solar_position=sun)
+
+        clearsky_poa = pvlib.irradiance.get_total_irradiance(self.pv_tilt, self.pv_azimuth, sun['apparent_zenith'],
+                                                             sun['azimuth'], clearsky['dni'], clearsky['ghi'],
+                                                             clearsky['dhi'], albedo=self.albedo, **kwargs)
+        clearsky_poa = clearsky_poa['poa_global']
+
+        if rescale is True:
+            clearsky_poa = normalization.irradiance_rescale(self.poa, clearsky_poa, method='iterative')
+
+        self.clearsky_poa = clearsky_poa
+
+    def calc_cell_temperature(self, poa, windspeed, ambient_temperature):
+        '''
+        Return cell temperature calculated from ambient conditions.
+
+        Parameters
+        ----------
+        poa = plane of array irradiance in W/m^2 (numeric)
+        windspeed = wind speed in m/s (numeric)
+        ambient_temperature = ambient temperature in Celsius (numeric)
+
+        Returns
+        -------
+        calculated cell temperature (numeric)
+        '''
+        if self.temperature_model is None:
+            cell_temp = pvlib.pvsystem.sapm_celltemp(poa, windspeed, ambient_temperature)
+        else:
+            cell_temp = pvlib.pvsystem.sapm_celltemp(poa, windspeed, ambient_temperature, model=self.temperature_model)
+        cell_temp = cell_temp['temp_cell']
+
+        return cell_temp
+
+    def calc_clearsky_tamb(self):
+        '''
+        Calculate clear-sky ambient temperature and store in self.clearsky_ambient_temperature
+        '''
+        times = self.clearsky_poa.index
+        if self.pvlib_location is None:
+            raise ValueError('pvlib location must be provided')
```

Comment:
This error message could be misinterpreted as telling the user to pass a pvlib location to this function instead of the original constructor of the SystemAnalysis object.  Maybe something like `'Missing location info, must pass pvlib_location to SystemAnalysis()'` and/or make a note in the docstring about the requirement?

Reply:
#233 addressed this